### PR TITLE
[Developer] Add "-add-help-link" parameter to kmcomp

### PR DESCRIPTION
**Pull request by darcywong00**
From review of sillsdev/keyman-keyman.com#220

kmcomp needs to handle a parameter ~`-addHelpPath`~ `-add-help-link` which populates .keyboard_info with a link to a keyboard's help page on https://help.keyman.com/keyboard/

### CODE REVIEW

File: windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas [hunk 1]
```diff
@@ -775,6 +779,18 @@ procedure TMergeKeyboardInfo.CheckOrAddMinKeymanVersion;
     json.AddPair('minKeymanVersion', MinVersionString);
 end;
 
+//
+// Add helpPath, from commandline parameter
+//
+procedure TMergeKeyboardInfo.AddHelpPath;
+
```
Comment (mcdurdin) [inline windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas]: delete blank line

File: windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas [hunk 2]
```diff
@@ -775,6 +779,18 @@ procedure TMergeKeyboardInfo.CheckOrAddMinKeymanVersion;
...
+begin
+    if FHelpPath = '' then
```
Comment (mcdurdin) [inline windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas]: only 2 space indent on this line

File: windows/src/developer/kmcomp/main.pas [hunk 3]
```diff
@@ -176,6 +177,11 @@ procedure Run;
       Inc(i);
       FParamSourcePath := ParamStr(i);
     end
+    else if s = '-addhelppath' then
```
Comment (mcdurdin) [inline windows/src/developer/kmcomp/main.pas]: Let's use `-add-help-path` to match the style of other parameters

**Comment by darcywong00** [comment]
TODOs: Investigate why the following links aren't being generated (content exists on help.keyman.com/keyboard)

## fv
- [ ] *_kmw

## greek_polytonic
- [ ] greek_polytonic_ls
- [ ] greek_polytonic_ls_ez
- [ ] greek_polytonic_ls_mt
- [ ] greek_polytonic_ms
- [ ] greek_polytonic_ms_ez
- [ ] greek_polytonic_ms_mt
- [ ] greek_polytonic_sa
- [ ] greek_polytonic_sa_ez
- [ ] greek_polytonic_sa_mt
- [ ]  greek_polytonic_unicode
- [ ] greek_polytonic_sa

## other
- [ ] indic roman transliteration
- [ ] ndailu keyman keyboard

- [ ] PhoneticMalayalam (keyboard ID is phoneticmalayalam)

- [ ] sipon phonetic sinhala
- [ ] urdu ph - package

File: windows/src/developer/kmcomp/main.pas (hunk 3 above)
Comment (darcywong00) [reply windows/src/developer/kmcomp/main.pas]: ack, I realized in api.keyman.com, we used `helpLink`. Now to s/helpPath/helpLink/g and `-add-help-link` accordingly

File: windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas [hunk 4]
```diff
@@ -775,6 +779,30 @@ procedure TMergeKeyboardInfo.CheckOrAddMinKeymanVersion;
...
+procedure TMergeKeyboardInfo.AddHelpLink;
+var
+  v: TJSONValue;
+
+begin
+  // Validate pre-existing helpLink
+  v := json.GetValue('helpLink');
+  if v <> nil then
+  begin
+    if v.Value.IndexOf('https://help.keyman.com/keyboard/') < 0 then
```
Comment (mcdurdin) [inline windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas]: This should be `if ... <> 0`

File: windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas]: chomp chomp still blank line to delete :)

Comment (darcywong00) [reply windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas]: Hmm, I wonder if GitHub was caching the diff. 
When the procedure was named `AddHelpPath` it had the extra line.

It **should** be gone now...

File: windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas (hunk 4 above)
Comment (darcywong00) [reply windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas]: I want this to raise when `IndexOf()` returns -1 because the string isn't found,

**Pull request merged**